docs/development.md

```diff
@@ -227,7 +227,30 @@ experimental-features = nix-command flakes
 
 ## Quick Usage Guide
 
-Get a Nix development shell:
+### With `direnv`
+
+The recommended way to develop is to install [`direnv`](https://direnv.net/)
+(usually via `apt` or any other package manager), then `cd` to the repository
+root, and run `direnv allow` in the terminal (only needs to be done the first
```

Comment:
So I tried these instructions, but nothing really happened.

What is the supposed output from `direnv allow`?

Reply:
something like

```console
$ direnv allow
direnv: loading ~/src/oak/.envrc
direnv: using flake
direnv: export +ANDROID_HOME +AR +AR_FOR_BUILD +AS +AS_FOR_BUILD +CC +CC_FOR_BUILD +CLASSPATH +CONFIG_SHELL +CXX +CXX_FOR_BUILD +GRADLE_OPTS +HOST_PATH +IN_NIX_SHELL +JAVA_HOME +LD +LD_FOR_BUILD +NIX_BINTOOLS +NIX_BINTOOLS_FOR_BUILD +NIX_BINTOOLS_WRAPPER_TARGET_BUILD_x86_64_unknown_linux_gnu +NIX_BINTOOLS_WRAPPER_TARGET_HOST_x86_64_unknown_linux_gnu +NIX_BUILD_CORES +NIX_BUILD_TOP +NIX_CC +NIX_CC_FOR_BUILD +NIX_CC_WRAPPER_TARGET_BUILD_x86_64_unknown_linux_gnu +NIX_CC_WRAPPER_TARGET_HOST_x86_64_unknown_linux_gnu +NIX_CFLAGS_COMPILE +NIX_CFLAGS_COMPILE_FOR_BUILD +NIX_ENFORCE_NO_NATIVE +NIX_HARDENING_ENABLE +NIX_LDFLAGS +NIX_LDFLAGS_FOR_BUILD +NIX_STORE +NM +NM_FOR_BUILD +NODE_PATH +OBJCOPY +OBJCOPY_FOR_BUILD +OBJDUMP +OBJDUMP_FOR_BUILD +PERL5LIB +RANLIB +RANLIB_FOR_BUILD +READELF +READELF_FOR_BUILD +SIZE +SIZE_FOR_BUILD +SOURCE_DATE_EPOCH +STRINGS +STRINGS_FOR_BUILD +STRIP +STRIP_FOR_BUILD +TEMP +TEMPDIR +TMP +TMPDIR +__structuredAttrs +buildInputs +buildPhase +builder +cmakeFlags +configureFlags +depsBuildBuild +depsBuildBuildPropagated +depsBuildTarget +depsBuildTargetPropagated +depsHostHost +depsHostHostPropagated +depsTargetTarget +depsTargetTargetPropagated +doCheck +doInstallCheck +dontAddDisableDepTrack +mesonFlags +name +nativeBuildInputs +out +outputs +patches +phases +preferLocalBuild +propagatedBuildInputs +propagatedNativeBuildInputs +shell +shellHook +stdenv +strictDeps +system ~PATH ~XDG_DATA_DIRS
```

what do you see?

Reply:
```
[10:26:52] myusername@myhost:~/oak$ direnv allow
[10:26:56] myusername@myhost:~/oak$ 
```

nothing, but 
```
$ direnv --help
direnv v2.32.1
```

Reply:
What does `direnv status` say in the folder?

Here is mine for reference

```console
$ direnv status
direnv exec path /nix/store/2srybdg8f4w206saap42bnz0vvihbrb1-direnv-2.32.3/bin/direnv
DIRENV_CONFIG /home/tzn/.config/direnv
bash_path /nix/store/mh1c3p23xby3chc3q0c0g78fa1nn65yw-bash-5.2-p15/bin/bash
disable_stdin false
warn_timeout 5s
whitelist.prefix []
whitelist.exact map[]
Loaded RC path /home/tzn/src/oak/.envrc
Loaded watch: ".envrc" - 2023-06-27T15:45:43+01:00
Loaded watch: "../../.local/share/direnv/allow/8d155404db807783092e8d33f1477ccb6a3c5b6ca48905f9bb7a4665d8f09dcd" - 2023-06-28T11:23:25+01:00
Loaded watch: "flake.nix" - 2023-06-28T11:25:31+01:00
Loaded watch: "flake.lock" - 2023-06-27T15:43:52+01:00
Loaded RC allowed false
Loaded RC allowPath
Found RC path /home/tzn/src/oak/.envrc
Found watch: ".envrc" - 2023-06-27T15:45:43+01:00
Found watch: "../../.local/share/direnv/allow/8d155404db807783092e8d33f1477ccb6a3c5b6ca48905f9bb7a4665d8f09dcd" - 2023-06-28T11:23:25+01:00
Found RC allowed true
Found RC allowPath /home/tzn/.local/share/direnv/allow/8d155404db807783092e8d33f1477ccb6a3c5b6ca48905f9bb7a4665d8f09dcd
```

Reply:
Anyways, apart from the output (maybe your shell has a different way of logging), does it work? i.e. do you have access to the binaries from the nix shell afterwards?

Reply:
No, it does no work:

```
[10:33:32] mschett@mschett:~/oak$ direnv status
direnv exec path /usr/bin/direnv
...
```

and 
```
[10:34:24] mschett@mschett:~/oak$ which rustc
/usr/local/google/home/mschett/.cargo/bin/rustc
```

Reply:
Can you paste the rest of `direnv status`?

Reply:
BTW depending on how you installed it and your shell config, you may need to also follow this maybe: https://direnv.net/docs/hook.html#bash (I didn't have to, but if that's necessary I'll add it to the instructions, assuming it solves the issue)

Reply:
I'll do so offline, I am never sure what to safely paste here.